v2/terraform_flannel.py

```diff
@@ -250,32 +253,50 @@ def _copyFrom(self, src, dest, machine, windows=False, root=False):
             self.logging.error("Ansible failed to fetch file from %s with error: %s" % (machine, out))
             raise Exception("Ansible failed to fetch file from %s with error: %s" % (machine, out))
 
+    # Used by _runRemoteCmd to add more retries
+    def _parseAnsibleOutput(self, out):
+        json_data = json.loads(out)
+        machinesToRetry = []
+        for vm in json_data['stats'].keys():
+            if json_data['stats'][vm]['failures'] == 1 or json_data['stats'][vm]['unreachable'] == 1:
+                machinesToRetry.append(vm)
+        return machinesToRetry
+
     def _runRemoteCmd(self, command, machines, windows=False, root=False):
         self.logging.info("Running cmd %s on remote machines %s." % (command, machines))
-        cmd=["ansible"]
-        if root:
-            cmd.append("--become")
-        if windows:
-            task = "win_shell"
-        else:
-            task = "shell"
-            cmd.append("--key-file=%s" % self.opts.ssh_private_key_path)
-
-        cmd.append("'%s'" % " ".join(machines))
-        cmd.append("-m")
-        cmd.append(task)
-        cmd.append("-a")
-        cmd.append("'%s'" % command)
 
-        ret, out = self._waitForConnection(machines, windows=windows)
-        if ret != 0:
-            self.logging.error("No connection to machines. Error: %s" % out)
-            raise Exception("No connection to machines. Error: %s" % out)
+        def _runRemoteAnsible(command, machines):
```

Comment:
What about the other parameters? windows, root etc? Shouldn't they be passed to this inner function as well?